src/org/zaproxy/zap/extension/proxies/OptionsProxiesPanel.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.proxies;
+
+import java.awt.GridBagLayout;
+import java.util.List;
+
+import javax.swing.JCheckBox;
+import javax.swing.JOptionPane;
+import javax.swing.SortOrder;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.option.SecurityProtocolsPanel;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.view.AbstractParamPanel;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.view.AbstractMultipleOptionsTablePanel;
+import org.zaproxy.zap.view.LayoutHelper;
+
+public class OptionsProxiesPanel extends AbstractParamPanel {
+
+    private static final long serialVersionUID = 1L;
+
+    private ProxiesMultipleOptionsPanel proxiesOptionsPanel;
+
+    private OptionsLocalProxyPanel mainProxyPanel;
+    private SecurityProtocolsPanel securityProtocolsPanel;
+    private OptionsProxiesTableModel proxiesModel;
+    private String currentAddress;
+    private int currentPort;
+
+    private ExtensionProxies extension;
+
+    public OptionsProxiesPanel(ExtensionProxies extension) {
+        super();
+        this.extension = extension;
+        initialize();
+    }
+
+    private void initialize() {
+        this.setName(Constant.messages.getString("proxies.options.title"));
+        this.setLayout(new GridBagLayout());
+        this.add(getMainProxyPanel(), LayoutHelper.getGBC(0, 0, 1, 0.0));

```

Comment:
We should probably add a scroll panel (to `OptionsProxiesPanel`), in my case it's starting to hide the table buttons (e.g. "Disable All").